sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasStateServer.scala

```diff
@@ -166,73 +170,69 @@ class TransformWithStateInPandasStateServer(
 
   private[sql] def handleValueStateRequest(message: ValueStateCall): Unit = {
     val stateName = message.getStateName
+    if (!valueStates.contains(stateName)) {
+      logWarning(log"Value state ${MDC(LogKeys.STATE_NAME, stateName)} is not initialized.")
+      sendResponse(1, s"Value state $stateName is not initialized.")
+      return
+    }
     message.getMethodCase match {
       case ValueStateCall.MethodCase.EXISTS =>
-        if (valueStates.contains(stateName) && valueStates(stateName).exists()) {
+        if (valueStates(stateName)._1.exists()) {
           sendResponse(0)
         } else {
           sendResponse(1, s"state $stateName doesn't exist")
```

Comment:
How do we distinguish the case of "no value state is defined for the state variable name" vs "the value state is defined but not having a value yet" if we use the same status code?